### DISCUSSION TITLE
altsrc for MultiStringFlag

### DISCUSSION THREAD

**Discussion by nkuba** [urfave/cli#1443] [Q&A]
Is it possible to read with `altsrc` the `MultiStringFlag` value to a destination?

My destination is `[]string`, so I wrapped `StringSliceFlag` with `MultiStringFlag`.
```
NetworkPeersFlag = &cli.MultiStringFlag{
	Target: &cli.StringSliceFlag{
		Name:  "network.peers",
	},
	Destination: &clientConfig.Peers,
}
```

There is no dedicated function in `altsrc` for `MultiStringFlag` and setting it with `altsrc.NewStringSliceFlag(NetworkPeersFlag.Target)` doesn't work.

**Comment by joeycumines** [comment]
Mmm ok, so I implemented the `cli.SliceFlag` wrapper (note: `MultiStringFlag = SliceFlag[*StringSliceFlag, []string, string]`), but I have not used the `altsrc` package myself, so I don't necessarily have all the context here.

tl;dr it looks like an issue with the `altsrc` package, and it doesn't appear to work correctly with `cli.StringSliceFlag.Destination`, either

The way `cli.SliceFlag` works is by utilising `flag.Value.Set` as a hook. It appears that `altsrc.StringSliceFlag` works by replacing the original `flag.Value`, in the flag set, which presumably only works for the `cli.Context.Get` (etc) mechanism.
https://github.com/urfave/cli/blob/c8147a48450b8842fce36b9d8a565149a14b1c51/altsrc/flag.go#L94
This behavior seems silly, but I'm guessing it was added due to the "append" behavior of the (base) `flag.Value.Set` implementations, used for all the slice flags.

Anyhow, on closer inspection, it looks like it probably doesn't work properly for the `Destination` field of the base `cli.StringSliceFlag`, either. Unfortunately, the way the different sources for flag values hang together is rather convoluted, and the `Destination` semantics for the slice flag implementations in particular have only recently worked "correctly", in terms of core functionality (and not considering altsrc).
